### PR TITLE
2017wk37

### PR DESCRIPTION
**Pull request by kerolasa**
Couple of compiler warning fixes, memory allocation checks, and an update to license file.

### CODE REVIEW

File: ipcalc-reverse.c [hunk 1]
```diff
@@ -97,7 +98,7 @@ static char hexchar(unsigned int val)
 	abort();
 }
 
-char *calc_reverse_dns6(struct in6_addr *ip, unsigned prefix)
+extern char *calc_reverse_dns6(struct in6_addr *ip, unsigned prefix)
```
Comment (nmav) [inline ipcalc-reverse.c]: Why do you add the extern keyword to these functions? It is also unrelated to the commit message.

File: ipcalc.c [hunk 2]
```diff
@@ -546,6 +547,7 @@ char *ipv4_prefix_to_hosts(char *hosts, unsigned hosts_size, unsigned prefix)
 	return hosts;
 }
 
+extern
```
Comment (nmav) [inline ipcalc.c]: I believe the addition of extern here is superfluous. That's the default for C functions. The static keyword that you have added, on the other hand is the right thing to do. Would it make sense to keep the static changes in this commit?

**Comment by nmav** [comment]
Seems fine except for the comments noted.

File: ipcalc.c (hunk 2 above)
Comment (kerolasa) [reply ipcalc.c]: I like code to be explicit what is visibility of functions, but sure not a big deal. Updated commit no longer has externs here and there.

**Pull request merged**